src/domains/ecommerce-common/03_storage.tf

```
@@ -391,17 +384,100 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "ecommerce_transient_enqu
   enabled        = true
   query = format(<<-QUERY
     let OpCountForQueue = (operation: string, queueKey: string) {
-      StorageQueueLogs
-      | where OperationName == operation and ObjectKey startswith queueKey
-      | summarize count()
+        StorageQueueLogs
+        | where OperationName == operation and ObjectKey startswith queueKey
+        | summarize count() 
+        | project count_ 
+        | extend dummy=1
+    };
+    let PutMessages = (queueName: string) {
+      OpCountForQueue("PutMessage", queueName)
+        | project PutCount = count_
+        | extend dummy = 1
+    };
+    let DeletedMessages =  (queueName: string) {
+      OpCountForQueue("DeleteMessage", queueName)
+        | project DeleteCount = count_
+        | extend dummy = 1
     };
     let MessageRateForQueue = (queueKey: string) {
-      OpCountForQueue("PutMessage", queueKey)
-      | join kind=fullouter OpCountForQueue("DeleteMessage", queueKey) on count_
-      | project name = queueKey, Count = count_ - count_1
+        PutMessages(queueKey)
+        | join kind=inner (DeletedMessages(queueKey)) on dummy
+        | extend Diff = PutCount - DeleteCount
+        | project PutCount, DeleteCount, Diff
     };
     MessageRateForQueue("%s")
-    | where Count > ${each.value.threshold}
+    | where Diff > ${each.value.threshold}
+    QUERY
+    , "/${module.ecommerce_storage_transient.name}/${local.project}-${each.value.queue_key}"
+  )
+  severity    = each.value.severity
+  frequency   = each.value.frequency
+  time_window = each.value.time_window
+  trigger {
+    operator  = "GreaterThan"
+    threshold = 0
+  }
+}
+
+locals {
+  queue_expiration_alert_props = var.env_short == "p" ? [
+    {
+      "queue_key"   = "transactions-expiration-queue"
+      "severity"    = 1
+      "time_window" = 15
+      "frequency"   = 15
+      "threshold"   = 40
+    },
+  ] : []
+}
+
+# Queue size: Ecommerce - ecommerce queues enqueues rate alert
+resource "azurerm_monitor_scheduled_query_rules_alert" "ecommerce_expiration_enqueue_rate_alert" {
```

```suggestion
resource "azurerm_monitor_scheduled_query_rules_alert" "ecommerce_enqueue_rate_alert_visibility_timeout_diff" {
```

make this alert name general